internal/common/utils.go

```diff
@@ -280,3 +281,28 @@ func EnsureAnyAdminHasNamespaceAccess(
 	}
 	return fmt.Errorf("no users have admin access to namespace %s in cluster %s", namespace, cluster)
 }
+
+// FetchSecretFromApplication retrieves the secret associated with the destination cluster of the given Application.
+func FetchSecretFromApplication(ctx context.Context, k8sClient client.Client, app *argoprojv1alpha1.Application) (*corev1.Secret, error) {
```

Comment:
```suggestion
FetchSecretFromApplication
```
FetchDestinationClusterSecret